core/src/main/java/nu/yona/server/subscriptions/service/UserDto.java

```diff
@@ -228,10 +229,21 @@ public static UserDto createInstanceWithPrivateData(User userEntity, Set<BuddyDt
 				userEntity.getMobileNumber(), userEntity.isMobileNumberConfirmed(), userEntity.getNamedMessageSourceId(),
 				userEntity.getAnonymousMessageSourceId(),
 				UserAnonymizedDto.getGoalsIncludingHistoryItems(userEntity.getAnonymized()), buddies,
-				userEntity.getUserAnonymizedId(), VPNProfileDto.createInstance(userEntity),
+				userEntity.getUserAnonymizedId(), createVpnProfileDto(userEntity),
 				userEntity.getDevices().stream().map(UserDeviceDto::createInstance).collect(Collectors.toSet()));
 	}
 
+	// YD-541: Remove this method
+	private static Optional<VPNProfileDto> createVpnProfileDto(User userEntity)
+	{
+		return determineDefaultDevice(userEntity).map(VPNProfileDto::createInstance);
+	}
+
+	private static Optional<UserDevice> determineDefaultDevice(User userEntity)
+	{
+		return userEntity.getDevices().stream().filter(d -> d.getDeviceAnonymized().getDeviceIndex() == 0).findAny();
```

Comment:
Isn't this a bit duplicate with `DeviceService.getDefaultDeviceId`?

Reply:
Yes. We could refactor the code now and move the logic to determine the default from ``DeviceService`` to ``UserDto``, but as soon as we address YD-541, we would need to move it back again. That's why I duplicated it here for a moment.

Reply:
OK, so this code will be removed later on.